### PR TITLE
DropDownList: dataSource should be updated even if there are selected items (T932182)

### PR DESCRIPTION
**Pull request by ksercs**
https://supportcenter.devexpress.com/internal/ticket/details/T932182

### CODE REVIEW

File: js/ui/tag_box.js [hunk 1]
```diff
@@ -1259,7 +1259,7 @@ const TagBox = SelectBox.inherit({
     },
 
     _refreshSelected: function() {
-        this._list && this._list.option('selectedItems', this._selectedItems);
+        this._list && this._list.getDataSource() && this._list.option('selectedItems', this._selectedItems);
```
Comment (DokaRus) [inline js/ui/tag_box.js]:
```suggestion
        this._list?.getDataSource() && this._list.option('selectedItems', this._selectedItems);
```
Touché!

File: js/ui/tag_box.js [hunk 2]
```diff
@@ -1370,7 +1370,11 @@ const TagBox = SelectBox.inherit({
     },
 
     _setListDataSource: function() {
+        const valueBackup = this._getValue();
```
Comment (DokaRus) [inline js/ui/tag_box.js]:
```suggestion
        const currentValue = this._getValue();
```

Comment (ksercs) [reply js/ui/tag_box.js]: fixed

File: js/ui/tag_box.js (hunk 1 above)
Comment (ksercs) [reply js/ui/tag_box.js]: fixed

**Review by DokaRus — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**